### ISSUE TITLE
In serve, after reload, the page is sometimes blank

### ISSUE THREAD

**Issue by dylanvdmerwe** [driftyco/ionic-app-scripts#338]
#### Short description of the problem:
Working on files while `ionic serve` is running, when the app detects changes it is meant to reload the page in the browser. However, every so often, after making changes to some html files, when the app reloads, the browser does reload the page but the page is blank. From the console logs it looks like something is happening, but nothing is shown on the page. I have to then force refresh the page to get things to display again. 

![image](https://cloud.githubusercontent.com/assets/2852839/20151175/e32717ec-a6c1-11e6-8efa-b625b65d4f0e.png)

```
JS changed:   www/build/main.js
CSS changed:  www/build/main.css
[21:14:03]  template update started ...
[21:14:03]  build update started ...
[21:14:03]  transpile update started ...
[21:14:03]  transpile update finished in 230 ms
[21:14:03]  webpack update started ...
[21:14:08]  webpack update finished in 4.74 s
[21:14:08]  build update finished in 4.97 s
[21:14:08]  template update finished in 4.97 s

[21:14:08]  watch ready
```

#### What behavior are you expecting?
The app should properly reload when changes are detected.

**Which @ionic/app-scripts version are you using?**
Cordova CLI: 6.4.0
Ionic Framework Version: 2.0.0-rc.2
Ionic CLI Version: 2.1.4
Ionic App Lib Version: 2.1.2
Ionic App Scripts Version: 0.0.41
ios-deploy version: 1.9.0
ios-sim version: 5.0.10
OS: Mac OS X Sierra
Node Version: v6.9.1
Xcode version: Xcode 8.1 Build version 8B62

**Comment by dylanvdmerwe** [comment]
Seems to be fixed with version 0.42.

**Comment by danbucholtz** [comment]
@dylanvdmerwe,

I think this is just a race condition of saving too many files back to back. It may have triggered multiple builds/bundle updates. We are now debouncing our file saves to prevent this.

I'm going to close this for now but I'll re-open if the issue crops up again. I think we have another change we'll make to make this even more rock solid.

Thanks,
Dan
